pkg/controller/job/job_controller.go

```diff
@@ -1070,6 +1077,12 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 		job.Status.Succeeded = int32(succeededIndexes.total())
 		job.Status.CompletedIndexes = succeededIndexes.String()
 	}
+	if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
+		if finishedCond != nil && finishedCond.Type == batch.AlphaNoCompatGuaranteeJobFailing {
+			job.Status.Conditions = append(job.Status.Conditions, *finishedCond)
```

Comment:
right here, you could set finishedCond again

Reply:
This place is yet before removal of the finalizers, so we should not be setting the JobFailed yet. 
We could potentially do it here (in addition to appending the interim FailureTarget condition): https://github.com/kubernetes/kubernetes/blob/aea9f9887d08922378b9e7da0e81a46e733f2995/pkg/controller/job/job_controller.go#L1120
but it would only be a performance improvement to save one cycle of syncJob. 
We would still need the interim FailureTarget job condition to account for scenario when job controller would crash between removal of finalizers and appending the final Failed condition (scenario now covered with an integration test).

Reply:
what I mean is that you could change the `finishedCond` to the final form right after line 1082.

This way, the final JobStatusUpdate can set the final finishedCond.

In any case, `enactJobFinished` might need tweaking (otherwise we might update the metric twice, once for each sync.

Reply:
Updated, also don't use the FailureTarget for the path without finalizers.